eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -179,11 +180,49 @@ case class Commitments(channelId: ByteVector32,
    * and our HTLC success in case of a force-close.
    */
   def almostTimedOutIncomingHtlcs(blockheight: Long, fulfillSafety: CltvExpiryDelta): Set[UpdateAddHtlc] = {
-    def nearlyExpired(add: UpdateAddHtlc) = blockheight >= (add.cltvExpiry - fulfillSafety).toLong
+    def nearlyExpired(add: UpdateAddHtlc): Boolean = blockheight >= (add.cltvExpiry - fulfillSafety).toLong
 
     localCommit.spec.htlcs.collect(incoming).filter(nearlyExpired)
   }
 
+  /** Compute our dust exposure in our commit tx (local) and their commit tx (remote). */
+  def currentDustExposure(): (MilliSatoshi, MilliSatoshi) = {
+    val localCommitDustExposure = CommitmentSpec.dustExposure(localCommit.spec, localParams.dustLimit, commitmentFormat)
+    val remoteCommitDustExposure = CommitmentSpec.dustExposure(remoteCommit.spec, remoteParams.dustLimit, commitmentFormat)
+    (localCommitDustExposure, remoteCommitDustExposure)
+  }
+
+  /** Test whether the given incoming htlc contributes to the dust exposure in the local or remote commit tx. */
```

Comment:
No, the description is correct. We really need to know if this incoming HTLC contributes to the dust exposure, to fail it instead of forward it if it overflows our maximum threshold.

Reply:
You're right that there's something wrong here: this function is used for both incoming and outgoing htlcs (because in both cases we make the same kind of checks). I need to change that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1985/commits/d0ffd73c1c1c49ceb1fd7c1b789fbfa9067da98f
I'll also update the unit test to make sure that it fails with the previous version of the code in a separate commit.

Reply:
I added an additional test in https://github.com/ACINQ/eclair/pull/1985/commits/feffafd9989782755529bf05f45ed770e2f8b661